File: Examples/test-suite/php/arrays_scope_runme.php
<?php

require "tests.php";

// No new functions
check::functions(array());
// New classes
check::classes(array('arrays_scope','Bar'));
// No new globals
check::globals(array());

$bar=new bar();
$bar->blah($bar->adata, $bar->bdata, $bar->cdata);
// Like C/C++, SWIG treats `int asize[ASIZE]` as `int*` so there's no checking
// of the passed array size.
$bar->blah($bar->bdata, $bar->cdata, $bar->adata);

check::done();

File: Examples/test-suite/perl5/arrays_scope_runme.pl
use strict;
use warnings;
use Test::More tests => 4;
BEGIN { use_ok('arrays_scope') }
require_ok('arrays_scope');

my $obj = arrays_scope::Bar->new();
isa_ok($obj, 'arrays_scope::Bar');
can_ok($obj, 'blah');

$obj->blah($obj->{adata}, $obj->{bdata}, $obj->{cdata});
